flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/TaskExecutorResourceUtils.java

```diff
@@ -668,24 +679,23 @@ MemorySize getTotalJvmMetaspaceAndOverheadSize() {
 		}
 	}
 
-	public static Configuration adjustConfigurationForLocalExecution(final Configuration configuration, int numberOfTaskExecutors) {
-
-		final long jvmFreeHeapMemBytesPerTm = EnvironmentInformation.getSizeOfFreeHeapMemoryWithDefrag() / numberOfTaskExecutors;
-		final MemorySize totalHeapMemory = new MemorySize(jvmFreeHeapMemBytesPerTm);
-
-		final MemorySize frameworkOffHeap = getFrameworkOffHeapMemorySize(configuration);
-		final MemorySize taskOffHeap = getTaskOffHeapMemorySize(configuration);
-		final MemorySize totalFlinkMemoryExceptShuffleAndManaged = totalHeapMemory.add(frameworkOffHeap).add(taskOffHeap);
-
-		final double shuffleFraction = getShuffleMemoryRangeFraction(configuration).fraction;
-		final double managedFraction = getManagedMemoryRangeFraction(configuration).fraction;
+	public static Configuration adjustConfigurationForLocalExecution(final Configuration configuration) {
+		final Configuration modifiedConfig = new Configuration(configuration);
 
-		final MemorySize estimatedTotalFlinkMemory = totalFlinkMemoryExceptShuffleAndManaged
-			.multiply(1 / (1 - shuffleFraction - managedFraction));
+		if (!isTaskExecutorResourceExplicitlyConfigured(configuration)) {
+			// This does not affect the JVM heap size for local execution,
+			// we simply set it to pass the sanity checks in memory calculations
+			modifiedConfig.set(TaskManagerOptions.TASK_HEAP_MEMORY, MemorySize.parse("100m"));
+		}
 
-		final Configuration modifiedConfig = new Configuration(configuration);
-		modifiedConfig.set(TaskManagerOptions.TOTAL_FLINK_MEMORY, estimatedTotalFlinkMemory);
+		if (!isShuffleMemoryExplicitlyConfigured(configuration)) {
+			modifiedConfig.set(TaskManagerOptions.SHUFFLE_MEMORY_MIN, MemorySize.parse("64m"));
+			modifiedConfig.set(TaskManagerOptions.SHUFFLE_MEMORY_MAX, MemorySize.parse("64m"));
+		}
 
+		if (!isManagedMemorySizeExplicitlyConfigured(configuration)) {
+			modifiedConfig.set(TaskManagerOptions.MANAGED_MEMORY_SIZE, MemorySize.parse("16m"));
+		}
```

Comment:
It's true that this method is only relevant for the `MiniCluster`.
However, there are similar demands for overwriting memory configs of a given configuration  instance, for other purposes (see #10716). I think it might be clearer to keep these similar methods at one place.

Reply:
Which other class will use this method in the foreseeable future? Once it becomes necessary to use this method at different places I'm ok with making it public. However, at the moment, this should only be used by the `MiniCluster` and, hence, let's move it there. Moreover, this would further increase code cohesion.

Reply:
Another reason to move this method is that we are making `TaskExecutorResourceUtils` responsible for another concern. `TaskExecutorResourceUtils` should not be concerned with overwriting configurations imo.